mowgli_etl/pipeline/wdc/wdc_size_buckets.py

```diff
@@ -0,0 +1,149 @@
+from typing import Tuple
+from mowgli_etl.pipeline.wdc.wdc_product_type import WdcProductType
+from mowgli_etl.pipeline.wdc.wdc_product_dimensions import WdcProductDimensions
+from mowgli_etl.pipeline.wdc.wdc_generic_size import WdcGenericSize
+
+
+class WdcSizeBuckets:
+    def __init__(self):
+        self.averages = dict()
+
+    def __bucket(self, dimension):
+        # Determine heuristics for buckets
+        return None
+
+    def __convert(self, dimension, origin):
+        if origin in (
+            "in",
+            "lb",
+            "lbs",
+            "v",
+        ):
+            return dimension
+        if origin == "mm":
+            return dimension * 3.9
+        if origin == "cm":
+            return dimension * 0.39
+        if origin == "m":
+            return dimension * 0.0039
+        if origin == "ft":
+            return dimension * 12
+        if origin == "mv":
+            return dimension / 1000
+        if origin == "kv":
+            return dimension * 1000
+        if origin == "oz":
+            return dimension / 16
+        if origin == "g":
+            return dimension * 0.0022046
+        if origin == "mg":
+            return dimension * 0.0000022046
+        if origin == "kg":
+            return dimension * 2.2046
+
+    def __update(self, generic):
+        if generic.name not in self.averages:
+            return generic
+        new_object = {
+            "depth": None,
+            "height": None,
+            "length": None,
+            "width": None,
+            "power": None,
+            "weight": None,
+        }
+        local = self.averages[generic.name]
+        for a in ("depth", "height", "length", "width", "power", "weight"):
+            if getattr(generic.dimension, a) is not None:
+                new_object[a] = {"value": None, "unit": None}
+                new_object[a]["value"] = (
+                    getattr(generic.dimension, a).value
+                    + getattr(local.dimension, a).value * local.count
+                ) / (local.count + 1)
+                new_object[a]["unit"] = getattr(generic.dimension, a).unit
+
+        updated_generic = {
+            "name": generic.name,
+            "dimension": new_object,
+            "count": local.count + 1,
+            "unit": generic.unit,
+        }
+
+        return WdcGenericSize.from_dict(updated_generic)
+
+    def generalize(self, wdc_product: Tuple[WdcProductType, WdcProductDimensions]):
+        # General units: in^3, V, lb
+        product_type = wdc_product[0]
+        product_dimensions = wdc_product[1]
+        dimension = None
+        unit = None
+        if (
```

Comment:
The if blocks here are essentially all the same. You can iterate over the fields of product_dimensions. Look in the `dataclasses` module. Something like
```
for field in dataclasses.fields(product_dimensions):
    dimension = getattr(product_dimensions, field.name)
    if not isinstance(dimension, WdcProductDimension):
       continue # Also takes care of None case
    dimension[field.name] = ...
```